simtools/applications/validate_file_using_schema.py

```diff
@@ -85,6 +104,18 @@ def _validate_ecsv_file(args_dict):
         data_file=args_dict["file_name"],
     )
     data_validator.validate_and_transform()
+    logger.debug("Successful validation of escv file")
+
+
+def _validate_metadata(args_dict, logger):
+    """
+    Validate metadata.
+
+    """
+
+    # metadata_collector runs the metadata validation by default, no need to do anything else
```

Comment:
An idea: I think it would be better if the other functions (`_validate_yaml_or_json_file`) could get also as compact as this one. Ideally, we move the functionalities (of `_validate_yaml_or_json_file` in this case) to the modules and keep the application clean.

Reply:
OK, this was anyway planned (and already implemented in another context and branch on my local disk). So I've moved now a function into the data_reader called `def read_value_from_file(file_name, schema_file=None, validate=False, metadata_file=None):`, which does exactly the same as `read_table_from_file` (with the obvious difference that it is for a value). It also includes the validation.

This simplifies that method.

Reply:
ok! Thanks. That is perfectly fine.